nixos/modules/misc/nixpkgs.nix

```diff
@@ -195,13 +198,10 @@ in
     };
 
     hostPlatform = lib.mkOption {
-      type = lib.types.either lib.types.str lib.types.attrs; # TODO utilize lib.systems.parsedPlatform
+      type = lib.types.either lib.types.str lib.types.attrs;
       example = {
         system = "aarch64-linux";
       };
-      # Make sure that the final value has all fields for sake of other modules
-      # referring to this. TODO make `lib.systems` itself use the module system.
-      apply = lib.systems.elaborate;
```

Comment:
Please add this back (or something like it), so that the type of `config.nixpkgs.hostPlatform` is always a platform attrset.